larpandoracontent/LArMonitoring/EventClusterValidationAlgorithm.h

```diff
@@ -0,0 +1,125 @@
+/**
+ *  @file   larpandoracontent/LArMonitoring/EventClusterValidationAlgorithm.h
+ *
+ *  @brief  Header file of the event-level cluster validation.
+ *
+ *  $Log: $
+ */
+#ifndef LAR_EVENT_CLUSTER_VALIDATION_ALGORITHM_H
+#define LAR_EVENT_CLUSTER_VALIDATION_ALGORITHM_H 1
+
+#include "Pandora/Algorithm.h"
+
+namespace lar_content
+{
+
+/**
+ *  @brief  EventClusterValidationAlgorithm class
+ */
+class EventClusterValidationAlgorithm : public pandora::Algorithm
+{
+private:
+    struct ClusterMetrics
+    {
+        ClusterMetrics();
+
+        std::vector<int> m_nRecoHits;
+        std::vector<float> m_purities;
+        std::vector<float> m_completenesses;
+        int m_nHits;
+        int m_nClusters;
+        int m_nMainMCs;
+    };
+
+    enum ValidationType
+    {
+        ALL,
+        SHOWER,
+        TRACK
+    };
+
+public:
+    struct CaloHitParents
+    {
+        CaloHitParents();
+
+        const pandora::MCParticle *m_pMainMC;
+        const pandora::Cluster *m_pCluster;
+    };
+
+    /**
+    *  @brief  Default constructor
+    */
+    EventClusterValidationAlgorithm();
+
+    virtual ~EventClusterValidationAlgorithm();
```

Comment:
Base Process destructor is already virtual, so don't want to redeclare as virtual here.